src/test/java/edu/kis/vh/nursery/RhymersJUnitTest.java

```diff
@@ -38,13 +86,75 @@ public void testIsFull() {
 		}
 
 		boolean result = rhymer.isFull();
-		Assert.assertEquals(true, result);
+		if(rhymer.stack instanceof IntArrayStack) Assert.assertEquals(true, result);
+	}
+	
+	@Test
+	public void testIsFullWithArray() {
+		DefaultCountingOutRhymer rhymer = new DefaultCountingOutRhymer(new IntArrayStack());
+		final int STACK_CAPACITY = 12;
+		for (int i = 0; i < STACK_CAPACITY-1; i++) {
+			boolean result = rhymer.isFull();
+			Assert.assertEquals(false, result);
+			rhymer.countIn(888);
+		}
+
+		boolean result = rhymer.isFull();
+		if(rhymer.stack instanceof IntArrayStack) Assert.assertEquals(true, result);
+	}
+	
+	@Test
+	public void testIsFullWithLinkedList() {
+		DefaultCountingOutRhymer rhymer = new DefaultCountingOutRhymer(new IntLinkedListAdapter(new IntLinkedList()));
+		final int STACK_CAPACITY = 12;
+		for (int i = 0; i < STACK_CAPACITY; i++) {
+			boolean result = rhymer.isFull();
+			Assert.assertEquals(false, result);
+			rhymer.countIn(888);
+		}
+
+		boolean result = rhymer.isFull();
+		if(rhymer.stack instanceof IntArrayStack) Assert.assertEquals(true, result);
 	}
 
 	@Test
 	public void testPeekaboo() {
 		DefaultCountingOutRhymer rhymer = new DefaultCountingOutRhymer();
-		final int EMPTY_STACK_VALUE = -1;
+		final int EMPTY_STACK_VALUE = 0;

```

Comment:
The original constant should be used (it would be best to use a common constant in a stacks interface).